internal/rlimit.go

```diff
@@ -0,0 +1,54 @@
+package internal
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/cilium/ebpf/internal/unix"
+)
+
+func RemoveMemlockRlimit() error {
+	var oldLimit unix.Rlimit
+	if err := unix.Prlimit(0, unix.RLIMIT_MEMLOCK, nil, &oldLimit); err != nil {
+		return fmt.Errorf("retrieve RLIMIT_MEMLOCK: %s", err)
+	}
+
+	// Reduce the limit to zero. This is always allowed.
+	newLimit := unix.Rlimit{Cur: 0, Max: oldLimit.Max}
```

Comment:
I'm a bit worried about a bad interaction with package `features`. This might cause false negatives on feature probes, as those might be run during package `init()` as well, though not necessarily in parallel.

If we're keeping the 'zero' behaviour around (see comments below), calling this `zeroLimit` is likely more appropriate.

Reply:
I think the answer is: `features` should not run things on init. Where are we doing that?

Reply:
I mean a caller using feature probe output in an `init()` function anywhere in a program.